src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EventPublisherContainer.java

```diff
@@ -40,11 +41,11 @@ public EventPublisher getPublisher(
                 CaseDetails caseDetails = caseRetrieval.get();
 
                 return caseDetails == null
-                    ? exceptionRecordCreator
-                    : new DelegatePublisher(attachDocsPublisher, caseDetails.getCaseTypeId());
+                    ? envelope -> exceptionRecordCreator.publish(envelope)
+                    : envelope -> attachDocsPublisher.publish(envelope, caseDetails.getCaseTypeId());
```

Comment:
_or_ could make a single lambda like

```java
return envelope -> caseDetails.get()
    ? exceptionRecordCreator.publish(envelope)
    : attachDocsPublisher.publish(envelope, caseDetails.getCaseTypeId());
```

Reply:
Not a bad idea with retrieving the case in the publisher. This way the method would just get the publisher, as advertised, without taking extra steps. That might affect exception handling, though.